recipes/yaclib/all/conanfile.py

```diff
@@ -0,0 +1,159 @@
+        **_yaclib_flags,
+        **_yaclib_faults,
+    }
+
+    default_options = {
+        'fPIC': True,
+        **{k: False for k in _yaclib_flags.keys()},
+        **{k: False for k in _yaclib_faults.keys()},
+    }
+
+    _cmake_definitions = {}
+
+    @property
+    def _source_subfolder(self):
+        return 'source_subfolder'
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            'gcc': '7',
+            'Visual Studio': '14.20',
+            'clang': '8',
+            'apple-clang': '12',
+        }
+
+    def _patch_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            patch(self, **p)
+
+    @lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        defs = {}
+        if not self.conf.get(
+                'tools.build:skip_test', default=True, check_type=bool):
+            defs['YACLIB_TEST'] = 'ON'
+            defs['BUILD_TESTING'] = 'ON'
+
+        if self.settings.compiler.get_safe('cppstd'):
+            defs['YACLIB_CXX_STANDARD'] = self.settings.compiler.cppstd
+
+        for definition, value in self._cmake_definitions.items():
+            defs[definition] = value
+
+        cmake.configure(build_script_folder=self._source_subfolder, variables=defs)
+        return cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        flags_definitions_list = []
+        for flag in self._yaclib_flags:
+            if self.options.get_safe(flag):
+                flags_definitions_list.append(flag.upper())
+
+        if flags_definitions_list:
+            self._cmake_definitions['YACLIB_FLAGS'] = ';'.join(
+                flags_definitions_list)
+
+        faults_definitions_list = []
+        for fault in self._yaclib_faults:
+            if self.options.get_safe(fault):
+                definition = fault.replace('_fault', '').upper()
+                faults_definitions_list.append(definition)
+
+        if faults_definitions_list:
+            self._cmake_definitions['YACLIB_FAULT'] = ';'.join(
+                faults_definitions_list)
+
+    def validate(self):
+        if self.settings.compiler.get_safe('cppstd'):
+            check_min_cppstd(self, 17)
+        else:
+            if self._compilers_minimum_version.get(str(self.settings.compiler)):
+                if Version(self.settings.compiler.version) < self._compilers_minimum_version.get(str(self.settings.compiler)):
+                    raise ConanInvalidConfiguration(
+                        "yaclib requires a compiler supporting c++17")
+            else:
+                self.output.warn(
+                    "yaclib recipe does not recognize the compiler. yaclib requires a compiler supporting c++17. Assuming it does.")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+        deps = CMakeDeps(self)
```

Comment:
Didnt see a requires method so you can remove this I think

Reply:
Removed deps